crates/biome_js_analyze/src/lint/nursery/no_adjacent_overload_signatures.rs

```diff
@@ -0,0 +1,332 @@
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_js_syntax::{
+    AnyJsClassMember, AnyJsModuleItem, AnyJsStatement, JsExport, JsIdentifierBinding,
+    JsModuleItemList, JsSyntaxKind, SyntaxNodeText, TsMethodSignatureTypeMember,
+};
+use biome_rowan::{AstNode, TextRange};
+
+declare_rule! {
+    /// Disallow non adjacent overload signatures.
```

Comment:
I'll be honest: "non-adjacent overload signatures" is a veeeery technical, and I doubt beginners will understand what this is about. Remember that as per our [philosophy](https://biomejs.dev/internals/philosophy/#technical), we want to try to use a simple vocabulary: 

> Reduce jargon. Don’t assume that users will understand specific terminology. Strive to provide precise meaning for experts and beginners. For example, use “character” where you would traditionally use “token” when producing parser errors.